sda-commons-server-opentelemetry/src/main/java/org/sdase/commons/server/opentelemetry/servlet/TracingFilter.java

```diff
@@ -0,0 +1,132 @@
+package org.sdase.commons.server.opentelemetry.servlet;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.sdase.commons.server.opentelemetry.decorators.ServerSpanDecorator;
+
+public class TracingFilter implements Filter {
```

Comment:
No test class found.

The implementation is rather complex. Is there no provided alternative by the library that is close to our implementation?